Index: logview4net/src/App/Viewers/textViewer.cs
===================================================================
--- logview4net/src/App/Viewers/textViewer.cs	(revision 376)
+++ logview4net/src/App/Viewers/textViewer.cs	(working copy)
@@ -106,6 +106,8 @@
         private ToolStripMenuItem findToolStripMenuItem;
         private ToolStripMenuItem addHideActionToolStripMenuItem;
         private ToolStripMenuItem showHiddenToolStripMenuItem;
+        private ToolStripMenuItem saveToFileToolStripMenuItem;
+        private SaveFileDialog saveFileDialog1;
 
         private IContainer components;
 
@@ -190,6 +192,7 @@
             this.components = new System.ComponentModel.Container();
             this.mnuPop = new System.Windows.Forms.ContextMenuStrip(this.components);
             this.addIgnoreActionToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.addHideActionToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
             this.addPopupActionToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
             this.addHighlightActionToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
             this.blueToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
@@ -209,10 +212,11 @@
             this.yellowToolStripMenuItem1 = new System.Windows.Forms.ToolStripMenuItem();
             this.toolStripSeparator1 = new System.Windows.Forms.ToolStripSeparator();
             this.findToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.showHiddenToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.saveToFileToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
             this.progress = new System.Windows.Forms.ProgressBar();
+            this.saveFileDialog1 = new System.Windows.Forms.SaveFileDialog();
             this.txt = new AdvRichTextBox();
-            this.addHideActionToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
-            this.showHiddenToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
             this.mnuPop.SuspendLayout();
             this.SuspendLayout();
             // 
@@ -226,7 +230,8 @@
             this.addHighlightmatchActionToolStripMenuItem,
             this.toolStripSeparator1,
             this.findToolStripMenuItem,
-            this.showHiddenToolStripMenuItem});
+            this.showHiddenToolStripMenuItem,
+            this.saveToFileToolStripMenuItem});
             this.mnuPop.Name = "mnuPop";
             this.mnuPop.Size = new System.Drawing.Size(223, 186);
             this.mnuPop.Opening += new System.ComponentModel.CancelEventHandler(this.mnuPop_Opening);
@@ -238,6 +243,13 @@
             this.addIgnoreActionToolStripMenuItem.Text = "Add &Ignore action";
             this.addIgnoreActionToolStripMenuItem.Click += new System.EventHandler(this.addIgnoreActionToolStripMenuItem_Click);
             // 
+            // addHideActionToolStripMenuItem
+            // 
+            this.addHideActionToolStripMenuItem.Name = "addHideActionToolStripMenuItem";
+            this.addHideActionToolStripMenuItem.Size = new System.Drawing.Size(222, 22);
+            this.addHideActionToolStripMenuItem.Text = "Add Hide action";
+            this.addHideActionToolStripMenuItem.Click += new System.EventHandler(this.addHideActionToolStripMenuItem_Click);
+            // 
             // addPopupActionToolStripMenuItem
             // 
             this.addPopupActionToolStripMenuItem.Name = "addPopupActionToolStripMenuItem";
@@ -262,49 +274,49 @@
             // blueToolStripMenuItem
             // 
             this.blueToolStripMenuItem.Name = "blueToolStripMenuItem";
-            this.blueToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.blueToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.blueToolStripMenuItem.Text = "&Blue";
             this.blueToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // cyanToolStripMenuItem
             // 
             this.cyanToolStripMenuItem.Name = "cyanToolStripMenuItem";
-            this.cyanToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.cyanToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.cyanToolStripMenuItem.Text = "&Cyan";
             this.cyanToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // limeToolStripMenuItem
             // 
             this.limeToolStripMenuItem.Name = "limeToolStripMenuItem";
-            this.limeToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.limeToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.limeToolStripMenuItem.Text = "&Lime";
             this.limeToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // magentaToolStripMenuItem
             // 
             this.magentaToolStripMenuItem.Name = "magentaToolStripMenuItem";
-            this.magentaToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.magentaToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.magentaToolStripMenuItem.Text = "&Magenta";
             this.magentaToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // orangeToolStripMenuItem
             // 
             this.orangeToolStripMenuItem.Name = "orangeToolStripMenuItem";
-            this.orangeToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.orangeToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.orangeToolStripMenuItem.Text = "&Orange";
             this.orangeToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // redToolStripMenuItem
             // 
             this.redToolStripMenuItem.Name = "redToolStripMenuItem";
-            this.redToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.redToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.redToolStripMenuItem.Text = "&Red";
             this.redToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
             // yellowToolStripMenuItem
             // 
             this.yellowToolStripMenuItem.Name = "yellowToolStripMenuItem";
-            this.yellowToolStripMenuItem.Size = new System.Drawing.Size(152, 22);
+            this.yellowToolStripMenuItem.Size = new System.Drawing.Size(121, 22);
             this.yellowToolStripMenuItem.Text = "&Yellow";
             this.yellowToolStripMenuItem.Click += new System.EventHandler(this.colorHighlightToolStripMenuItem_Click);
             // 
@@ -383,6 +395,21 @@
             this.findToolStripMenuItem.Text = "&Find";
             this.findToolStripMenuItem.Click += new System.EventHandler(this.findToolStripMenuItem_Click);
             // 
+            // showHiddenToolStripMenuItem
+            // 
+            this.showHiddenToolStripMenuItem.Enabled = false;
+            this.showHiddenToolStripMenuItem.Name = "showHiddenToolStripMenuItem";
+            this.showHiddenToolStripMenuItem.Size = new System.Drawing.Size(222, 22);
+            this.showHiddenToolStripMenuItem.Text = "Show hidden";
+            // 
+            // saveToFileToolStripMenuItem
+            // 
+            this.saveToFileToolStripMenuItem.Name = "saveToFileToolStripMenuItem";
+            this.saveToFileToolStripMenuItem.ShortcutKeys = ((System.Windows.Forms.Keys)((System.Windows.Forms.Keys.Control | System.Windows.Forms.Keys.S)));
+            this.saveToFileToolStripMenuItem.Size = new System.Drawing.Size(222, 22);
+            this.saveToFileToolStripMenuItem.Text = "Save to file ...";
+            this.saveToFileToolStripMenuItem.Click += new System.EventHandler(this.saveToFileToolStripMenuItem_Click);
+            // 
             // progress
             // 
             this.progress.BackColor = System.Drawing.Color.Black;
@@ -394,6 +421,10 @@
             this.progress.TabIndex = 1;
             this.progress.Visible = false;
             // 
+            // saveFileDialog1
+            // 
+            this.saveFileDialog1.Filter = "Text files (*.txt)|*.txt|Rich text files (*.rtf)|*.rtf";
+            // 
             // txt
             // 
             this.txt.BackColor = System.Drawing.Color.Black;
@@ -411,20 +442,6 @@
             this.txt.WordWrap = false;
             this.txt.KeyUp += new System.Windows.Forms.KeyEventHandler(this.txt_KeyUp);
             // 
-            // addHideActionToolStripMenuItem
-            // 
-            this.addHideActionToolStripMenuItem.Name = "addHideActionToolStripMenuItem";
-            this.addHideActionToolStripMenuItem.Size = new System.Drawing.Size(222, 22);
-            this.addHideActionToolStripMenuItem.Text = "Add Hide action";
-            this.addHideActionToolStripMenuItem.Click += new System.EventHandler(this.addHideActionToolStripMenuItem_Click);
-            // 
-            // showHiddenToolStripMenuItem
-            // 
-            this.showHiddenToolStripMenuItem.Enabled = false;
-            this.showHiddenToolStripMenuItem.Name = "showHiddenToolStripMenuItem";
-            this.showHiddenToolStripMenuItem.Size = new System.Drawing.Size(222, 22);
-            this.showHiddenToolStripMenuItem.Text = "Show hidden";
-            // 
             // textViewer
             // 
             this.BackColor = System.Drawing.Color.Black;
@@ -1096,5 +1113,24 @@
             }
         }
 
+        private void saveToFileToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (_log.Enabled) _log.Debug(GetHashCode(), "saveToFileToolStripMenuItem_Click");
+            //saveFileDialog1.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+//            saveFileDialog1.DefaultExt = ".txt";
+            saveFileDialog1.ShowDialog(this);
+            if (saveFileDialog1.FileName != "")
+            {
+                if (saveFileDialog1.FilterIndex == 2)
+                {
+                    txt.SaveFile(saveFileDialog1.FileName, RichTextBoxStreamType.RichText);
+                }
+                else
+                {
+                    txt.SaveFile(saveFileDialog1.FileName, RichTextBoxStreamType.UnicodePlainText);
+                }
+            }
+        }
+
     }
 }
Index: logview4net/src/App/Viewers/textViewer.resx
===================================================================
--- logview4net/src/App/Viewers/textViewer.resx	(revision 376)
+++ logview4net/src/App/Viewers/textViewer.resx	(working copy)
@@ -120,4 +120,7 @@
   <metadata name="mnuPop.TrayLocation" type="System.Drawing.Point, System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
     <value>17, 17</value>
   </metadata>
+  <metadata name="saveFileDialog1.TrayLocation" type="System.Drawing.Point, System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
+    <value>115, 17</value>
+  </metadata>
 </root>
\ No newline at end of file
